common/lib/xmodule/xmodule/library_tools.py

```diff
@@ -117,25 +117,37 @@ def update_children(self, dest_block, user_id, user_perms=None):
         if user_perms and not user_perms.can_write(dest_block.location.course_key):
             raise PermissionDenied()
 
-        new_libraries = []
+        if not dest_block.source_library_id:
+            dest_block.source_library_version = ""
+            return
+
         source_blocks = []
-        for library_key, __ in dest_block.source_libraries:
-            library = self._get_library(library_key)
-            if library is None:
-                raise ValueError("Required library not found.")
-            if user_perms and not user_perms.can_read(library_key):
-                raise PermissionDenied()
-            filter_children = (dest_block.capa_type != ANY_CAPA_TYPE_VALUE)
-            if filter_children:
-                # Apply simple filtering based on CAPA problem types:
-                source_blocks.extend([key for key in library.children if self._filter_child(key, dest_block.capa_type)])
-            else:
-                source_blocks.extend(library.children)
-            new_libraries.append(LibraryVersionReference(library_key, library.location.library_key.version_guid))
+        library_key = dest_block.source_library_key
+        library = self._get_library(library_key)
+        if library is None:
+            raise ValueError("Required library not found.")
```

Comment:
@bradenmacdonald I can't tell if this is an error message that is made visible to users, but wonder if it should be "*Requested* library not found."?